src/txn/txn_rollback_to_stable.c

```diff
@@ -1287,6 +1381,11 @@ __rollback_to_stable(WT_SESSION_IMPL *session)
     conn->stable_rollback_maxfile = conn->next_file_id + 1;
     WT_WITH_SCHEMA_LOCK(session, ret = __rollback_to_stable_btree_apply(session));
 
+err:
+    /* This is quite expensive, so don't waste work. */
+    if (exclusive)
```

Comment:
The logic of the flag to release may have problems. 
While processing the exclusive on mode, one handle is exclusive is turned on but the second one failed, this leads to never turn off the first one.

Reply:
Can you explain more? I don't understand what you mean.

Reply:
Oh, I think I understand. We should just remove this `exclusive` flag since I don't think we can do anything sensible here. We always need to release regardless.

Reply:
c289a0c

Reply:
@kommiharibabu Ok, there's still a problem here. We need to know which trees we successfully turned off eviction for so we know which trees to decrement `btree->evict_disabled`.

I'm going to wait til @agorrod looks at this and confirms the overall idea (switching off eviction for the check) before fixing that.